src/state.rs

```diff
@@ -28,6 +28,19 @@ use crate::modules::ModuleConfig;
 use crate::modules::{compile_module, HostModules};
 use crate::{Schedule, VMError};
 
+#[derive(Debug, Clone)]
```

Comment:
Why are you using pub fields here? It seems like allowing runtime modification of the name or origin should not be allowed.

Reply:
You're right. Went with "getters" instead.